spacy/matcher/dependencymatcher.pyx

```diff
@@ -423,6 +427,26 @@ cdef class DependencyMatcher:
     def _left_sib(self, doc, node):
         return [doc[child.i] for child in doc[node].head.children if child.i < node]
 
+    def _right_child(self, doc, node):
+        return [doc[child.i] for child in doc[node].children if child.i > node]
+    
+    def _left_child(self, doc, node):
+        return [doc[child.i] for child in doc[node].children if child.i < node]
+
+    def _right_parent(self, doc, node):
+        if doc[node].head == doc[node]:
```

Comment:
Do you need this condition? 

If the node is the root (= is its own head), then it should fail the position check, since equal values don't match.

Reply:
Sorry for getting back to this so late. 

> Do you need this condition?
> 
> If the node is the root (= is its own head), then it should fail the position check, since equal values don't match.

No, you are right! I have made these changes and pushed up to this branch.

To answer your questions regarding tests. I have had a look in the test py file you linked above. Having very little experience in testing, I am feeling nervous about writing a competent test. Maybe you could point to some resources I could have a read?  I'm quite happy to contribute to documentation where needed when I find my way around the documentation.

I have also made slight changes in how these REL_OPs are referred to. I used the way right_child and left_child are referred with "++" and "--" indicating positions. I think it is probably easily to remember?

Thanks!